juju/juju
# Make upgrade-gui a controller command.

Move the API client methods to the controller client.
Also implement a way to mock controller HTTP calls.

QA:
- bootstrap a controller;
- run `juju help upgrade-gui` and check that a -c (--controller) option is now available;
- run `juju upgrade-gui --list` and check that it properly returns a list of GUI archives available on simplestreams;
- run `juju upgrade-gui` to check that we already are at the most recent version (i.e. output should be "Juju GUI at version 2.1.10");
- run `juju upgrade-gui 2.1.8` to switch to an older version;
- run `juju gui` and check that in this old version the login form was a bit weird (with really large text inputs);
- run `juju upgrade-gui no-such`: it should output "ERROR invalid GUI release version or local path "no-such"";
- run `juju upgrade-gui 0.0.1`: it should output "ERROR Juju GUI release version 0.0.1 not found";
- download jujugui-2.1.11.tar.bz2 from https://github.com/juju/juju-gui/releases;
- run `juju upgrade-gui ~/path/to/jujugui-2.1.11.tar.bz2` and check that the upgrade suceeds;
- run `juju gui --show-credentials`: check that the login form is now rendered properly;
- log into the GUI and check that it works (for instance, deploy an application).


(Review request: http://reviews.vapour.ws/r/5619/)
